File: Strivers A2Z DSA Sheet/812-RotateString/812-RotateString.cpp
// Last updated: 8/7/2026, 7:11:11 PM
class Solution {
public:
    bool rotateString(string s, string goal) {
        if(s.size() != goal.size()) return false;

        string doubled = s+s;
        return doubled.find(goal) != string::npos;
    }
};

File: Strivers A2Z DSA Sheet/812-RotateString/812-RotateString.java
// Last updated: 8/7/2026, 7:10:46 PM
class Solution {
    public boolean rotateString(String s, String goal) {
        if(s.length() != goal.length()) return false;

        for(int i=0; i<s.length(); i++){
            String result = s.substring(i) + s.substring(0,i);
            if(result.equals(goal)) return true;
        }
        return false;
    }
}